extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -192,11 +179,27 @@ export class AmpAdApiHandler {
    * @private
    */
   updateSize_(height, width, source, origin) {
-    this.baseInstance_.attemptChangeSize(height, width).then(() => {
-      this.sendEmbedSizeResponse_(
-        true /* success */, width, height, source, origin);
-    }, () => this.sendEmbedSizeResponse_(
-        false /* success */, width, height, source, origin));
+    // Calculate new width and new height based on the requested size.
```

Comment:
How about this, might be shorter and clearer:
`Calculate new width and height of the container to include the padding. If padding is negative, just use the requested width and height directly.`